### PR TITLE
feat(lib/runtime/storage): Remove `sortedKeys` from `lib/runtime/storage`

### LINKED ISSUE #4075: Remove `sortedKeys` map usage from `lib/runtime/storage`
### Description

We should remove the usage of `sortedKeys` map from `lib/runtime/storage` because to keep it in sync with the current state when starting Gossamer from a snapshot would be required a dump from the keys in the state to the map. A better description can be found here: https://github.com/ChainSafe/gossamer/issues/4064#issuecomment-2210986268

### Acceptance Criteria

- Don't replace `sortedKeys` with `t.state.Entries()` given that the method `Entries()` does a heavy operation of retrieving all keys and values from the state, instead try to use iterators to query the state if needed

### PR DESCRIPTION
**Pull request by EclesioMeloJunior**
## Changes

- Remove the usage of `sortedKeys` from `lib/runtime/storage` and use the state if needed

## Tests

N/A

## Issues
- Depends on https://github.com/ChainSafe/gossamer/pull/4073
- Closes https://github.com/ChainSafe/gossamer/issues/4075

### CODE REVIEW

**Comment by dimartiro** [comment]
Since this PR depends on #4073 what if we change the base to `eclesio/in-memory-trie-iterator` ?

**Review by dimartiro — APPROVED** [review]
Great cleanup!

File: pkg/trie/inmemory/iterator.go [hunk 1]
```diff
@@ -46,24 +46,24 @@ func NewInMemoryTrieIterator(opts ...IterOpts) *InMemoryTrieIterator {
 	return iter
 }
 
-func (t *InMemoryTrieIterator) NextEntry() *trie.Entry {
-	found := findNextNode(t.trie.root, []byte(nil), t.cursorAtKey)
+func (i *InMemoryTrieIterator) NextEntry() *trie.Entry {
```
Comment (kishansagathiya) [inline pkg/trie/inmemory/iterator.go]: renaming t to i feels meh!

**Review by kishansagathiya — APPROVED** [review]
Went on a call with @EclesioMeloJunior to review this quicker. He helped me understand this better, so thanks!

**Pull request merged**